src/brsTypes/components/RoSGNode.ts

```diff
@@ -1189,9 +1189,18 @@ function addChildren(interpreter: Interpreter, node: RoSGNode, typeDef: Componen
         if (newChild instanceof RoSGNode && appendChild) {
             appendChild.call(interpreter, newChild);
             let setField = newChild.getMethod("setfield");
-            for (let [key, value] of Object.entries(child.fields)) {
-                if (setField) {
-                    setField.call(interpreter, new BrsString(key), new BrsString(value));
+            if (setField) {
```

Comment:
Good optimization.  Is this check still needed though?  When does a type of RoSGNode not have _setField_ implemented?  I see several similar checks in the file that I don't understand.  @alimnios72 , do you know why they're needed?

Reply:
The check was added because `getMethod` can return `undefined` https://github.com/sjbarag/brs/blob/master/src/brsTypes/components/BrsComponent.ts#L40  and the typescript compiler won't let you use `setField` without a check